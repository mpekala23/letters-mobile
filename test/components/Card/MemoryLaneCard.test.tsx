import React from 'react';
import { MemoryLaneCard } from '@components';
import { fireEvent, render, toJSON } from '@testing-library/react-native';

const setup = (propOverrides = {}) => {
  const props = {
    text: 'Text',
    date: 'Date',
    imageUri: 'test',
    color: 'green',
    onPress: jest.fn(),
    ...propOverrides,
  };
  return {
    ...render(<MemoryLaneCard {...props} />),
    props,
  };
};

describe('Memory Lane Card component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should fire onPress() on a press', () => {
    const { props, getByText } = setup();
    fireEvent.press(getByText('Text'));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should display text', () => {
    const { getByText } = setup();
    expect(getByText('Text')).toBeDefined();
  });

  it('should display date', () => {
    const { getByText } = setup();
    expect(getByText('Date')).toBeDefined();
  });

  it('should display memory lane image', () => {
    const { getByTestId } = setup();
    expect(getByTestId('memoryLaneImage').props.source.uri).toBe('test');
  });
});
