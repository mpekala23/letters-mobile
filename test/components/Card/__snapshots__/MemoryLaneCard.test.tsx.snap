// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Memory Lane Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "height": 210,
          "padding": 0,
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
    testID="memoryLaneCard"
  >
    <Image
      source={
        Object {
          "uri": "test",
        }
      }
      style={
        Object {
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
          "height": "50%",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
      testID="memoryLaneImage"
    />
    <View
      style={
        Object {
          "paddingHorizontal": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "500",
              "overflow": "hidden",
            },
            Object {
              "height": 65,
            },
          ]
        }
      >
        Text
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6E757C",
              "fontSize": 16,
            },
            Object {
              "marginTop": 6,
            },
          ]
        }
      >
        Jul 12, 2019
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
